src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +23,28 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    private static final int LENGTH = 5;
+
+    public String getUniqueCharacters(String fileName) throws IOException {
+        Set<Character> letters = new TreeSet<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            while (reader.ready()) {
+                int character = reader.read();
+                if (Character.isLetter(character)) {
+                    letters.add((char) Character.toLowerCase(character));
+                }
+            }
+        }
+
+        StringBuilder result = new StringBuilder();
+        for (Character c : letters) {
+            result.append(c);
+            if (result.length() == LENGTH) {
+                break;
+            }
+        }
+        return result.length() < LENGTH
+                ? result.toString()
+                : result.toString().substring(0, LENGTH);
```

Comment:
do you need the ternary operator?) sometimes we need to set free someone we love)

Reply:
😭